public/pages/CreateTrigger/containers/CreateTrigger/utils/formikToTrigger.js

```diff
@@ -52,19 +75,37 @@ export function formikToCondition(values, monitorUiMetadata = {}) {
   const aggregationType = _.get(monitorUiMetadata, 'search.aggregationType', 'count');
 
   if (searchType === SEARCH_TYPE.QUERY) return { script: values.script };
+  if (searchType === SEARCH_TYPE.AD) return getADCondition(values);
   const isCount = aggregationType === 'count';
   const resultsPath = getResultsPath(isCount);
   const operator = getOperator(thresholdEnum);
   return getCondition(resultsPath, operator, thresholdValue, isCount);
 }
 
+export function getADCondition(values) {
+  const { anomalyDetector } = values;
+  if (anomalyDetector.triggerType === TRIGGER_TYPE.AD) {
+    const anomalyGradeOperator = getOperator(anomalyDetector.anomalyGradeThresholdEnum);
+    const anomalyConfidenceOperator = getOperator(anomalyDetector.anomalyConfidenceThresholdEnum);
+    return {
+      script: {
+        lang: 'painless',
+        //TODO:: Check null check is required or not confirm with backend
```

Comment:
Make issue for this if not being fixed in this PR

Reply:
Not an issue now, will remove this TODO.